scala/README.md

```diff
@@ -534,22 +610,71 @@ Tips & Tricks
 
     ```scala
     // Value classes
-    case class Email(email: String) extends AnyVal // no extra object allocation at runtime
+    case class Email(email: String) extends AnyVal // No extra object allocation at runtime
 
-    val email = Email("no@one.com") // never gets mixed with regular strings
+    val email = Email("joe@example.com") // Never gets mixed with other strings
 
     // Extension methods
     implicit class IntOps(val n: Int) extends AnyVal {
       def stars = "*" * n
     }
 
-    5.stars // equivalent to a static method call, no implicit conversion actually takes place
+    5.stars // Equivalent to a static method call, no implicit conversion actually takes place
     ```
 
 0. Leverage parallel collections, use `.par` judiciously.
 
 0. Use the `@tailrec` annotation to ensure the compiler can recognize a recursive method is tail-recursive.
 
+0. You can use the `@scala.beans.BeanProperty` and `@BooleanBeanProperty` annotations to automatically generate JavaBeans style getter and setter methods.
+
+0. Use this syntax to pass a sequence as parameter to a variable length argument list method:
+
+    ```scala
+    def foo(args: Int*) = ???
+
+    val seq = Seq(1, 2, 3)
+
+    foo(seq:_*)
+    ```
+
+0. There are some really neat tricks we can do with pattern matching:
+
+    ```scala
+    val tuple = ("foo", 1, false)
+    val (x, y, z) = tuple
+    assert((x, y, z) == (("foo", 1, false)))
+
+    case class Foo(x: String, y: Int)
+    val foo = Foo("foo", 1)
+    val Foo(a, b) = foo
+    assert((a, b) == (("foo", 1)))
+
+    val seq = Seq(1, 2, 3, 4, 5, 6)
+    val x :: xs = seq
+    assert((x, xs) == ((1, Seq(2, 3, 4, 5, 6))))
+
+    // Same as above
+    val Seq(y, ys@_*) = seq
+    assert((y, ys) == ((1, Seq(2, 3, 4, 5, 6))))
+
+    // Skipping elements
+    val _ :: a :: b :: _ :: zs = seq
+    assert((a, b, zs) == ((2, 3, Seq(5, 6))))
+
+    // Works with other collections, too
+    val vector = Vector(1, 2, 3, 4, 5, 6)
+    val Vector(_, a, b, _, ws@_*) = vector
+    assert((a, b, ws) == ((2, 3, Vector(5, 6))))
+
+    // Regular expressions
+    val regex = """(.)(.)(.)""".r
+    val regex(a, b, c) = "xyz" // Matches and extracts regex against "xyz"
+    assert((a, b, c) == (("x", "y", "z")))
+    ```
+
+0. Instead of running sbt tasks from your command shell (`$ sbt compile`, for instance), it is better to open an sbt prompt (just type `$ sbt`) and never leave it. Running all your sbt tasks (`clean`, `update`, `compile`, `test`, etc.) directly inside the sbt prompt is a lot faster since you only have to start sbt, load the JVM, and wait for it to warm up (if ever) once. If your `build.sbt` file changes, just run the `reload` task and you are good to go again.
```

Comment:
And if you want to do some sbt tasks and then leave it open, you can use the `shell` task, like `sbt clean update test:compile test shell` to do a clean full compile, test, and then have it leave you at the sbt prompt.